module/solutions/helper.js

```diff
@@ -1970,6 +1970,85 @@ module.exports = class SolutionsHelper {
         return reject(error);
       }
     });
+  }
+
+  /**
+  * Update User District and Organisation In Solutions For Reporting.
+  * @method
+  * @name addReportInformationInSolution 
+  * @param {String} solutionId - solution id.
+  * @param {Object} userProfile - user profile details
+  * @returns {Object} Solution information.
+*/
+
+  static addReportInformationInSolution(solutionId,userProfile) {
+    return new Promise(async (resolve, reject) => {
+        try {
+
+            //check solution & userProfile is exist
+            if ( solutionId && Object.keys(userProfile).length > 0 ) {
+                let district = [];
+                let organisation = [];
+
+                //get the districts from the userProfile
+                for (const location of userProfile["userLocations"]) {
+                    if ( location.type == messageConstants.common.DISTRICT ) {
+                        let distData = {}
+                        distData["locationId"] = location.id;
+                        distData["name"] = location.name;
+                        district.push(distData);
+                    }
+                }
+
+                //get the organisations from the userProfile
+                for (const org of userProfile["organisations"]) {
+                    let orgData = {};
+                    orgData.orgName = org.orgName;
+                    orgData.organisationId = org.organisationId;
+                    organisation.push(orgData);
+                }
+
+                //checking solution is exist
+                let solutionDocument = await this.solutionDocuments({
+                    _id: solutionId
+                },
+                ["_id"]);
+                
+                if( !solutionDocument.length > 0 ) {
+                    throw {
+                        message : CONSTANTS.apiResponses.SOLUTION_NOT_FOUND,
+                        status : HTTP_STATUS_CODE['bad_request'].status
+                    }
+                }
+
+                let updateQuery = {};
+                updateQuery["$addToSet"] = {};
+                updateQuery["$addToSet"]["reportInformation.organisations"] = { $each : organisation};
```

Comment:
@VISHNUDAS-tunerlabs , @priyanka-TL should we not check if there is any data in district or organisation or is it a blank array.